x-pack/plugins/index_lifecycle_management/public/sections/wizard/components/index_template/components/configuration/configuration.js

```diff
@@ -0,0 +1,179 @@
+
+import { ErrableFormRow } from '../../../../form_errors';
+import { NodeAttrsDetails } from '../../../node_attrs_details';
+
+export class Configuration extends Component {
+  static propTypes = {
+    fetchNodes: PropTypes.func.isRequired,
+    setSelectedNodeAttrs: PropTypes.func.isRequired,
+    setSelectedPrimaryShardCount: PropTypes.func.isRequired,
+    setSelectedReplicaCount: PropTypes.func.isRequired,
+    validate: PropTypes.func.isRequired,
+
+    selectedPrimaryShardCount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+    selectedNodeAttrs: PropTypes.string.isRequired,
+    nodeOptions: PropTypes.array.isRequired,
+    selectedReplicaCount: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+    isShowingErrors: PropTypes.bool.isRequired,
+    errors: PropTypes.object.isRequired,
+    isPrimaryShardCountHigherThanSelectedNodeAttrsCount: PropTypes.bool.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      isShowingNodeDetailsFlyout: false,
+    };
+  }
+
+  componentWillMount() {
+    this.props.fetchNodes();
+  }
+
+  render() {
+    const {
+      setSelectedNodeAttrs,
+      setSelectedPrimaryShardCount,
+      setSelectedReplicaCount,
+      validate,
+
+      nodeOptions,
+      selectedPrimaryShardCount,
+      selectedReplicaCount,
+      selectedNodeAttrs,
+      errors,
+      isShowingErrors,
+      isPrimaryShardCountHigherThanSelectedNodeAttrsCount,
+    } = this.props;
+
+    const primaryNodeErrors = isPrimaryShardCountHigherThanSelectedNodeAttrsCount ? (
+      <EuiCallOut
+        style={{ marginTop: '1rem' }}
+        title="Primary shard count is too high"
+        color="warning"
+        iconType="help"
+      >
+        The shard count should be lower than the number of nodes that match the selected attributes.
+      </EuiCallOut>
+    ) : null;
+
+    return (
+      <div>
+        <EuiDescribedFormGroup
+          title={<h4>Configure hot indices</h4>}
+          titleSize="s"
+          description="A hot index is actively being written to."
+          fullWidth
+        >
+          <ErrableFormRow
+            label="Choose where to allocate indices by node attribute"
+            errorKey={STRUCTURE_NODE_ATTRS}
+            isShowingErrors={isShowingErrors}
+            errors={errors}
+            helpText={
+              selectedNodeAttrs ? (
+                <EuiButtonEmpty
+                  flush="left"
+                  iconType="eye"
+                  onClick={() => this.setState({ isShowingNodeDetailsFlyout: true })}
+                >
+                  View a list of nodes attached to this configuration
+                </EuiButtonEmpty>
+              ) : null
+            }
+          >
+            <EuiSelect
+              value={selectedNodeAttrs || ' '}
+              onChange={async e => {
+                await setSelectedNodeAttrs(e.target.value);
+                validate();
```

Comment:
I expected this to call a prop, e.g. `onFieldChange`, provided by the consumer, which would then be responsible for updating the overall form state (possibly including validation) and passing the state back down as a prop into this component.

I'm finding it difficult to follow the existing data flow:

1. Dispatch async action from within leaf node (which doesn't seem to need to be async)
2. Actions updates nodes reducer with selected node attributes
3. Middleware intercepts this action and dispatches `setPhaseData` action configured with `PHASE_WARM` and `PHASE_NODE_ATTRS` enums.
4. It looks like this action is in turn intercepted by the middleware and dispatches `setPhaseData` but with the `PHASE_COLD` enum.
5. At this point I'm a little lost to be honest...  is it possible steps 3 and 4 aren't even needed, since it doesn't look like they affect the UI state? If the purposes of those steps are to update a payload object for the eventual "save" action, then I think it makes sense to perform that logic only when "save" is clicked.

Reply:
tabling these things for the moment as I think the whole thing will need to get reworked later.